asm/specialized_metadata.go

```diff
@@ -1359,15 +1359,28 @@ func (gen *generator) irDISubrange(new metadata.SpecializedNode, old *ast.DISubr
 		panic(fmt.Errorf("invalid IR specialized metadata node for AST specialized metadata node; expected *metadata.DISubrange, got %T", new))
 	}
 	for _, oldField := range old.Fields() {
+		var err error
 		switch oldField := oldField.(type) {
 		case *ast.CountField:
-			count, err := gen.irMDFieldOrInt(oldField.Count())
+			md.Count, err = gen.irMDFieldOrInt(oldField.Count())
```

Comment:
While this like many aspects is a matter of personal preference, for consistency, use the same `v, err :=` initialization as the other parts of this source file.

From https://github.com/llir/llvm/blob/b5d79ce650113f3c63805e4432b9a5113a18cbf0/asm/specialized_metadata.go#L129
```go
			scope, err := gen.irMDField(oldField.Scope())
			if err != nil {
				return nil, errors.WithStack(err)
			}
			md.Scope = scope
```

So, for these it would be.

```go
lowerBound, err := gen.irMDFieldOrInt(oldField.LowerBound())
if err != nil {
   return nil, errors.WithStack(err)
}
md.LowerBound = lowerBound
```

And likewise for `Stride` and `Count`.

I agree with you that both styles are equivalent in functionality and one may be fewer lines than the other for this case. The reason I prefer consistency is that it becomes especially important to have a consistent code base when working on a large project with more than one developer.

Hope that explains the rationale.

Cheers,
Robin